README.md

```diff
@@ -1,2 +1,218 @@
+Calculates the average distance between pairs of clusters.
+
+---
+
+### `minDistance(arrayA, arrayB, distanceMatrix)`
+
+Calculates the smallest distance between pairs of clusters.
+
+---
+
+### `maxDistance(arrayA, arrayB, distanceMatrix)`
+
+Calculates the largest distance between pairs of clusters.
+
+---
+
+### Comparison with [hcluster.js](https://github.com/cmpolis/hcluster.js)
+
+- This package does not include items from the original dataset in the results.
+Results are given in terms of indexes, either with respect to the original dataset or the distance matrix.
+- This package uses more modern JavaScript syntaxes and practices to make the code cleaner and simpler.
+- This package provides an `onProgress` callback and calls `postMessage` for use in [web workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers).
+Because clustering can take a long time with large data sets,
+you may want to run it as a web worker so the browser doesn't freeze for a long time, and you may need a callback so you can give users visual feedback on its progress.
+- This package makes some performance optimizations, such as removing unnecessary loops through big sets.
+It has been tested on modern OS's (Windows, Mac, Linux, iOS, Android), devices (desktop, laptop, mobile), browsers (Chrome, Firefox, Safari), contexts (main thread, web worker), and hosting locations (local, online).
+The results vary widely, and are likely sensitive to the specifics of hardware, cpu cores, browser implementation, etc.
+But in general, this package is significantly more performant than `hcluster.js`, to varying degrees, and is always at least as performant on average.
+Chrome seems to see the most performance gains (up to 10x, when the row number is high), while Firefox seems to see no gains.
+- This package does not touch the input data object, whereas the `hcluster.js` package does.
+D3 often expects you to mutate data objects directly, which is now typically considered bad practice in JavaScript.
+Instead, this package returns the useful data from the clustering algorithm (including the distance matrix), and allows you to mutate or not mutate the data object depending on your needs.
+In the future, a simple option could be added to instruct the algorithm to mutate the data object, if users can provide good use cases for what information is needed for constructing various D3 visualizations.
+
+---
+
+### Making changes to the library
+
+1. [Install yarn](https://classic.yarnpkg.com/en/docs/install)
+2. Clone this repo and navigate to it in your command terminal
+3. Run `yarn install`
+4. Make desired changes to `./src/hclust.js`
+5. Run `npm run test` to automatically rebuild the library and run test suite
+6. Run `npm run build` to just rebuild the library, and output the compiled contents to `./build/hclust.min.js`
+7. Commit changes to repo if necessary. *Make sure to run the build command before committing; it wont happen automatically.*
```

Comment:
Will add a section below this linking to other similar libraries:

https://github.com/cmpolis/hcluster.js/
https://github.com/harthur/clustering
https://github.com/mljs/hclust
https://github.com/math-utils/hierarchical-clustering